src/Azure.Functions.Cli/Actions/AzureActions/BaseAzureAction.cs

```diff
@@ -134,6 +141,16 @@ private async Task<string> GetAccessToken()
             throw new CliException($"Unable to connect to Azure. Make sure you have the `az` CLI or `{_azProfileModuleName}` PowerShell module installed and logged in and try again");
         }
 
+        private (bool succeeded, string token) TryGetTokenFromTestEnvironment()
```

Comment:
pretty minor nit: but unless your `Try...` method is async, you can use out parameters instead of a tuple of `(success, result)`. It's a bit more idiomatic in C#. 

```suggestion
        private bool TryGetTokenFromTestEnvironment(out string token)
```

then the usage becomes a bit clearer

```csharp
if (TryGetTokenFromTestEnvironment(out string token))
{
    return token;
}
```